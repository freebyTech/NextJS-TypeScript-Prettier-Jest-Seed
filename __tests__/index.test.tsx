import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import App from "../pages/index";

describe("With Enzyme", () => {
  it('App shows "My Blog"', () => {
    const app = shallow(<App />);

    expect(app.find("h1").text()).toEqual("My Blog");
  });
});

describe("With Snapshot Testing", () => {
  it('App shows "My Blog"', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
