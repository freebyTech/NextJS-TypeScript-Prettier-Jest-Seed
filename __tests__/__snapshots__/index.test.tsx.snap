// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "My Blog" 1`] = `
<div
  style={
    Object {
      "border": "1px solid #DDD",
      "margin": 20,
      "padding": 20,
    }
  }
>
  <div>
    <a
      href="/"
      onClick={[Function]}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      Home
    </a>
    <a
      href="/about"
      onClick={[Function]}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      About
    </a>
  </div>
  <h1>
    My Blog
  </h1>
  <ul>
    <li>
      <a
        href="/post?title=Hello Next.js"
        onClick={[Function]}
      >
        Hello Next.js
      </a>
    </li>
    <li>
      <a
        href="/post?title=Learn Next.js is awesome"
        onClick={[Function]}
      >
        Learn Next.js is awesome
      </a>
    </li>
    <li>
      <a
        href="/post?title=Deploy apps with Zeit"
        onClick={[Function]}
      >
        Deploy apps with Zeit
      </a>
    </li>
  </ul>
</div>
`;
